lib/metasploit/framework/login_scanner/smb.rb

```diff
@@ -135,7 +135,7 @@ def attempt_login(credential)
           rescue RubySMB::Error::UnexpectedStatusCode => e
             status = Metasploit::Model::Login::Status::INCORRECT
           ensure
-            client.disconnect!
+            client.disconnect! unless client.nil?
```

Comment:
This seems not very readable, how about `client.disconnect! if client`

Reply:
sure